SoftLayer/managers/registration.py

```diff
@@ -36,6 +36,92 @@ def detail(self, identifier):
             return self.registration.getObject(mask=mask, id=registration['id'])
         return None
 
+    def update_all(self, contact_id, skip_subnets=None):
+        """Updates every subnet to contact
+
+        :param list skip_subnets: A list of subnets ids to skip
+        :param int contact_id: Id of a Account_Regional_Registry_Detail that is a person
+        :return:A list of subnet registrations
+        """
+        skip_subnets = [] if skip_subnets is None else skip_subnets
+        subnets = self.get_public_subnets()
+        subnets = [subnet for subnet in subnets if subnet.get('id') not in skip_subnets]
+        new_subnets_registration = [subnet for subnet in subnets if subnet.get('activeRegistration') is None]
+        subnets_to_update = [subnet for subnet in subnets if subnet not in new_subnets_registration]
+        result = []
+        if new_subnets_registration:
+            templates = []
+            for subnet in new_subnets_registration:
+                new_registration = self.get_registration_template(contact_id, subnet)
+                templates.append(new_registration)
+            result = self.registration.createObjects(templates)
+
+        if subnets_to_update:
+            for subnet in subnets_to_update:
+                result.append(self.register_subnet(subnet, contact_id))
+
+        return result
+
+    def get_registration_template(self, contact_id, subnet):
+        """Gets registration template
+
+        :param int contact_id: Id of a Account_Regional_Registry_Detail that is a person
+        :param dict subnet: A subnet
+        :return:A registration template
+        """
+        return {
+            'networkIdentifier': subnet.get('networkIdentifier'),
+            'cidr': subnet.get('cidr'),
+            'detailReferences': [
+                {'detailId': contact_id}
+            ],
+        }
+
+    def get_public_subnets(self):
+        """Gets public subnets"""
+
+        mask = 'mask[id,networkIdentifier,activeRegistration,regionalInternetRegistry,cidr]'
+        subnets = self.account.getPublicSubnets(mask=mask)
+        return subnets
+
+    def register_subnet(self, subnet, contact_id):
```

Comment:
This function is just called `register` on my local branch, can you make sure you don't have two functions doing the same thing in your branch.

Reply:
updated